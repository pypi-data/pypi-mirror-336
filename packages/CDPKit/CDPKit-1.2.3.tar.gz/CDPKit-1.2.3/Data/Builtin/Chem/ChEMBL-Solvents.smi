WATER	[OH2]
DICHLOROMETHANE	ClCCl
TRICHLOROMETHANE	ClC(Cl)Cl
ETHYL ACETATE	CCOC(=O)C
METHANOL	CO
PROPAN-2-OL	CC(C)O
ACETONE	CC(=O)C
DMSO	CS(=O)C
ETHANOL	CCO
