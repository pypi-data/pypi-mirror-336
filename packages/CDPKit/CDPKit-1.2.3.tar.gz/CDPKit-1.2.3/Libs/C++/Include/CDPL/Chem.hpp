/* 
 * Chem.hpp 
 *
 * This file is part of the Chemical Data Processing Toolkit
 *
 * Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
 *
 * This library is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 2 of the License, or (at your option) any later version.
 *
 * This library is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this library; see the file COPYING. If not, write to
 * the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
 * Boston, MA 02111-1307, USA.
 */

/**
 * \file
 * \brief A convenience header including everything that is defined in namespace CDPL::Chem.
 */

#ifndef CDPL_CHEM_HPP
#define CDPL_CHEM_HPP

#include "CDPL/Chem/Entity3D.hpp"
#include "CDPL/Chem/Entity3DContainer.hpp"
#include "CDPL/Chem/AtomContainer.hpp"
#include "CDPL/Chem/BondContainer.hpp"
#include "CDPL/Chem/MolecularGraph.hpp"
#include "CDPL/Chem/Molecule.hpp"
#include "CDPL/Chem/Atom.hpp"
#include "CDPL/Chem/Bond.hpp"
#include "CDPL/Chem/Reaction.hpp"
#include "CDPL/Chem/BasicReaction.hpp"
#include "CDPL/Chem/BasicMolecule.hpp"
#include "CDPL/Chem/BasicAtom.hpp"
#include "CDPL/Chem/BasicBond.hpp"
#include "CDPL/Chem/Fragment.hpp"
#include "CDPL/Chem/ElectronSystem.hpp"
#include "CDPL/Chem/AtomDictionary.hpp"
#include "CDPL/Chem/FragmentList.hpp"
#include "CDPL/Chem/ElectronSystemList.hpp"
#include "CDPL/Chem/Entity3DMapping.hpp"
#include "CDPL/Chem/AtomMapping.hpp"
#include "CDPL/Chem/BondMapping.hpp"
#include "CDPL/Chem/AtomBondMapping.hpp"
#include "CDPL/Chem/StringDataBlock.hpp"
#include "CDPL/Chem/MatchConstraintList.hpp"
#include "CDPL/Chem/StereoDescriptor.hpp"
#include "CDPL/Chem/MultiConfMoleculeInputProcessor.hpp"
#include "CDPL/Chem/DefaultMultiConfMoleculeInputProcessor.hpp"
#include "CDPL/Chem/Atom3DCoordinatesFunction.hpp"
#include "CDPL/Chem/AtomPriorityFunction.hpp"
#include "CDPL/Chem/AtomPredicate.hpp"
#include "CDPL/Chem/BondPredicate.hpp"
#include "CDPL/Chem/AtomCompareFunction.hpp"
#include "CDPL/Chem/BondCompareFunction.hpp"
#include "CDPL/Chem/MatchExpression.hpp"
#include "CDPL/Chem/MatchExpressionList.hpp"
#include "CDPL/Chem/ANDMatchExpressionList.hpp"
#include "CDPL/Chem/ORMatchExpressionList.hpp"
#include "CDPL/Chem/NOTMatchExpression.hpp"
#include "CDPL/Chem/PropertyMatchExpression.hpp"
#include "CDPL/Chem/AtomTypeMatchExpression.hpp"
#include "CDPL/Chem/AtomEnvironmentMatchExpression.hpp"
#include "CDPL/Chem/AtomConfigurationMatchExpression.hpp"
#include "CDPL/Chem/AtomSSSRRingSizeMatchExpression.hpp"
#include "CDPL/Chem/BondConfigurationMatchExpression.hpp"
#include "CDPL/Chem/BondDirectionMatchExpression.hpp"
#include "CDPL/Chem/BondSubstituentDirectionMatchExpression.hpp"
#include "CDPL/Chem/BondReactionCenterStatusMatchExpression.hpp"
#include "CDPL/Chem/MolecularGraphComponentGroupingMatchExpression.hpp"
#include "CDPL/Chem/ReactionAtomMappingMatchExpression.hpp"
#include "CDPL/Chem/ReactionComponentGroupingMatchExpression.hpp"
#include "CDPL/Chem/Entity3DProperty.hpp"
#include "CDPL/Chem/AtomProperty.hpp"
#include "CDPL/Chem/BondProperty.hpp"
#include "CDPL/Chem/MolecularGraphProperty.hpp"
#include "CDPL/Chem/ReactionProperty.hpp"
#include "CDPL/Chem/AtomPropertyDefault.hpp"
#include "CDPL/Chem/BondPropertyDefault.hpp"
#include "CDPL/Chem/MolecularGraphPropertyDefault.hpp"
#include "CDPL/Chem/ReactionPropertyDefault.hpp"
#include "CDPL/Chem/AtomPropertyFlag.hpp"
#include "CDPL/Chem/BondPropertyFlag.hpp"
#include "CDPL/Chem/ControlParameter.hpp"
#include "CDPL/Chem/ControlParameterDefault.hpp"
#include "CDPL/Chem/ControlParameterFunctions.hpp"
#include "CDPL/Chem/AtomConfiguration.hpp"
#include "CDPL/Chem/BondConfiguration.hpp"
#include "CDPL/Chem/CIPDescriptor.hpp"
#include "CDPL/Chem/MDLParity.hpp"
#include "CDPL/Chem/BondStereoFlag.hpp"
#include "CDPL/Chem/AtomType.hpp"
#include "CDPL/Chem/SybylAtomType.hpp"
#include "CDPL/Chem/MOL2MoleculeType.hpp"
#include "CDPL/Chem/MOL2ChargeType.hpp"
#include "CDPL/Chem/HybridizationState.hpp"
#include "CDPL/Chem/ReactionCenterStatus.hpp"
#include "CDPL/Chem/ReactionRole.hpp"
#include "CDPL/Chem/BondDirection.hpp"
#include "CDPL/Chem/RadicalType.hpp"
#include "CDPL/Chem/INCHIReturnCode.hpp"
#include "CDPL/Chem/MDLDataFormatVersion.hpp"
#include "CDPL/Chem/DataFormat.hpp"
#include "CDPL/Chem/TautomerizationType.hpp"
#include "CDPL/Chem/AtomMatchConstraint.hpp"
#include "CDPL/Chem/BondMatchConstraint.hpp"
#include "CDPL/Chem/MolecularGraphMatchConstraint.hpp"
#include "CDPL/Chem/ReactionMatchConstraint.hpp"
#include "CDPL/Chem/INCHIMoleculeInputHandler.hpp"
#include "CDPL/Chem/INCHIMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/JMEMoleculeInputHandler.hpp"
#include "CDPL/Chem/JMEReactionInputHandler.hpp"
#include "CDPL/Chem/JMEMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/JMEReactionOutputHandler.hpp"
#include "CDPL/Chem/MOLMoleculeInputHandler.hpp"
#include "CDPL/Chem/MOLMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFMoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/RXNReactionInputHandler.hpp"
#include "CDPL/Chem/RXNReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFReactionInputHandler.hpp"
#include "CDPL/Chem/RDFReactionOutputHandler.hpp"
#include "CDPL/Chem/SMILESMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESReactionOutputHandler.hpp"
#include "CDPL/Chem/SMARTSMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMARTSReactionInputHandler.hpp"
#include "CDPL/Chem/SMARTSMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMARTSReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFMoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFReactionInputHandler.hpp"
#include "CDPL/Chem/CDFReactionOutputHandler.hpp"
#include "CDPL/Chem/MOL2MoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZMoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLMoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/SDFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESGZReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESGZReactionOutputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFGZReactionInputHandler.hpp"
#include "CDPL/Chem/RDFGZReactionOutputHandler.hpp"
#include "CDPL/Chem/RDFBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/RDFBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/CDFBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CDFGZReactionInputHandler.hpp"
#include "CDPL/Chem/CDFGZReactionOutputHandler.hpp"
#include "CDPL/Chem/CDFBZ2ReactionInputHandler.hpp"
#include "CDPL/Chem/CDFBZ2ReactionOutputHandler.hpp"
#include "CDPL/Chem/MOL2GZMoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2GZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/MOL2BZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/MOL2BZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/XYZBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/XYZBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLGZMoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLGZMolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/CMLBZ2MoleculeInputHandler.hpp"
#include "CDPL/Chem/CMLBZ2MolecularGraphOutputHandler.hpp"
#include "CDPL/Chem/SDFGZMoleculeReader.hpp"
#include "CDPL/Chem/SDFGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/SDFBZ2MoleculeReader.hpp"
#include "CDPL/Chem/SDFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/RDFGZReactionReader.hpp"
#include "CDPL/Chem/RDFGZReactionWriter.hpp"
#include "CDPL/Chem/RDFBZ2ReactionReader.hpp"
#include "CDPL/Chem/RDFBZ2ReactionWriter.hpp"
#include "CDPL/Chem/SMILESGZMoleculeReader.hpp"
#include "CDPL/Chem/SMILESGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESBZ2MoleculeReader.hpp"
#include "CDPL/Chem/SMILESBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESGZReactionReader.hpp"
#include "CDPL/Chem/SMILESGZReactionWriter.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionReader.hpp"
#include "CDPL/Chem/SMILESBZ2ReactionWriter.hpp"
#include "CDPL/Chem/CDFGZMoleculeReader.hpp"
#include "CDPL/Chem/CDFGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFBZ2MoleculeReader.hpp"
#include "CDPL/Chem/CDFBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFGZReactionReader.hpp"
#include "CDPL/Chem/CDFGZReactionWriter.hpp"
#include "CDPL/Chem/CDFBZ2ReactionReader.hpp"
#include "CDPL/Chem/CDFBZ2ReactionWriter.hpp"
#include "CDPL/Chem/MOL2GZMoleculeReader.hpp"
#include "CDPL/Chem/MOL2GZMolecularGraphWriter.hpp"
#include "CDPL/Chem/MOL2BZ2MoleculeReader.hpp"
#include "CDPL/Chem/MOL2BZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZGZMoleculeReader.hpp"
#include "CDPL/Chem/XYZGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZBZ2MoleculeReader.hpp"
#include "CDPL/Chem/XYZBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLGZMoleculeReader.hpp"
#include "CDPL/Chem/CMLGZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLBZ2MoleculeReader.hpp"
#include "CDPL/Chem/CMLBZ2MolecularGraphWriter.hpp"
#include "CDPL/Chem/JMEMoleculeReader.hpp"
#include "CDPL/Chem/JMEReactionReader.hpp"
#include "CDPL/Chem/JMEMolecularGraphWriter.hpp"
#include "CDPL/Chem/JMEReactionWriter.hpp"
#include "CDPL/Chem/SMILESMoleculeReader.hpp"
#include "CDPL/Chem/SMILESReactionReader.hpp"
#include "CDPL/Chem/SMILESMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMILESReactionWriter.hpp"
#include "CDPL/Chem/SMARTSMoleculeReader.hpp"
#include "CDPL/Chem/SMARTSReactionReader.hpp"
#include "CDPL/Chem/SMARTSMolecularGraphWriter.hpp"
#include "CDPL/Chem/SMARTSReactionWriter.hpp"
#include "CDPL/Chem/MOLMoleculeReader.hpp"
#include "CDPL/Chem/MOLMolecularGraphWriter.hpp"
#include "CDPL/Chem/SDFMoleculeReader.hpp"
#include "CDPL/Chem/SDFMolecularGraphWriter.hpp"
#include "CDPL/Chem/RXNReactionReader.hpp"
#include "CDPL/Chem/RXNReactionWriter.hpp"
#include "CDPL/Chem/RDFReactionReader.hpp"
#include "CDPL/Chem/RDFReactionWriter.hpp"
#include "CDPL/Chem/INCHIMoleculeReader.hpp"
#include "CDPL/Chem/INCHIMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFMoleculeReader.hpp"
#include "CDPL/Chem/CDFMolecularGraphWriter.hpp"
#include "CDPL/Chem/CDFReactionReader.hpp"
#include "CDPL/Chem/CDFReactionWriter.hpp"
#include "CDPL/Chem/MOL2MoleculeReader.hpp"
#include "CDPL/Chem/MOL2MolecularGraphWriter.hpp"
#include "CDPL/Chem/XYZMoleculeReader.hpp"
#include "CDPL/Chem/XYZMolecularGraphWriter.hpp"
#include "CDPL/Chem/CMLMoleculeReader.hpp"
#include "CDPL/Chem/CMLMolecularGraphWriter.hpp"
#include "CDPL/Chem/MoleculeReader.hpp"
#include "CDPL/Chem/MolecularGraphWriter.hpp"
#include "CDPL/Chem/ReactionReader.hpp"
#include "CDPL/Chem/ReactionWriter.hpp"
#include "CDPL/Chem/CompleteRingSet.hpp"
#include "CDPL/Chem/AromaticRingSet.hpp"
#include "CDPL/Chem/AromaticSSSRSubset.hpp"
#include "CDPL/Chem/SmallestSetOfSmallestRings.hpp"
#include "CDPL/Chem/ExtendedSSSR.hpp"
#include "CDPL/Chem/AromaticSubstructure.hpp"
#include "CDPL/Chem/CyclicSubstructure.hpp"
#include "CDPL/Chem/ConnectedSubstructureSet.hpp"
#include "CDPL/Chem/ComponentSet.hpp"
#include "CDPL/Chem/PiElectronSystemList.hpp"
#include "CDPL/Chem/SubstructureSearch.hpp"
#include "CDPL/Chem/ReactionSubstructureSearch.hpp"
#include "CDPL/Chem/CommonConnectedSubstructureSearch.hpp"
#include "CDPL/Chem/MaxCommonAtomSubstructureSearch.hpp"
#include "CDPL/Chem/MaxCommonBondSubstructureSearch.hpp"
#include "CDPL/Chem/AutomorphismGroupSearch.hpp"
#include "CDPL/Chem/MultiSubstructureSearch.hpp"
#include "CDPL/Chem/Reactor.hpp"
#include "CDPL/Chem/MorganNumberingCalculator.hpp"
#include "CDPL/Chem/CIPPriorityCalculator.hpp"
#include "CDPL/Chem/CIPConfigurationLabeler.hpp"
#include "CDPL/Chem/CanonicalNumberingCalculator.hpp"
#include "CDPL/Chem/HashCodeCalculator.hpp"
#include "CDPL/Chem/Atom2DCoordinatesCalculator.hpp"
#include "CDPL/Chem/Hydrogen3DCoordinatesCalculator.hpp"
#include "CDPL/Chem/BondStereoFlagCalculator.hpp"
#include "CDPL/Chem/BondOrderCalculator.hpp"
#include "CDPL/Chem/KekuleStructureCalculator.hpp"
#include "CDPL/Chem/SymmetryClassCalculator.hpp"
#include "CDPL/Chem/SurfaceAtomExtractor.hpp"
#include "CDPL/Chem/PatternAtomTyper.hpp"
#include "CDPL/Chem/SubstructureHistogramCalculator.hpp"
#include "CDPL/Chem/TautomerScore.hpp"
#include "CDPL/Chem/TautomerGenerator.hpp"
#include "CDPL/Chem/DefaultTautomerGenerator.hpp"
#include "CDPL/Chem/TautomerizationRule.hpp"
#include "CDPL/Chem/PatternBasedTautomerizationRule.hpp"
#include "CDPL/Chem/KetoEnolTautomerization.hpp"
#include "CDPL/Chem/ImineEnamineTautomerization.hpp"
#include "CDPL/Chem/NitrosoOximeTautomerization.hpp"
#include "CDPL/Chem/AmideImidicAcidTautomerization.hpp"
#include "CDPL/Chem/LactamLactimTautomerization.hpp"
#include "CDPL/Chem/KeteneYnolTautomerization.hpp"
#include "CDPL/Chem/NitroAciTautomerization.hpp"
#include "CDPL/Chem/PhosphinicAcidTautomerization.hpp"
#include "CDPL/Chem/SulfenicAcidTautomerization.hpp"
#include "CDPL/Chem/GenericHydrogen13ShiftTautomerization.hpp"
#include "CDPL/Chem/GenericHydrogen15ShiftTautomerization.hpp"
#include "CDPL/Chem/ResonanceStructureGenerator.hpp"
#include "CDPL/Chem/StereoisomerGenerator.hpp"
#include "CDPL/Chem/ChEMBLStandardizer.hpp"
#include "CDPL/Chem/ProtonationStateStandardizer.hpp"
#include "CDPL/Chem/FragmentGenerator.hpp"
#include "CDPL/Chem/RECAPRuleID.hpp"
#include "CDPL/Chem/RECAPAtomLabel.hpp"
#include "CDPL/Chem/RECAPFragmentGenerator.hpp"
#include "CDPL/Chem/BRICSRuleID.hpp"
#include "CDPL/Chem/BRICSAtomLabel.hpp"
#include "CDPL/Chem/BRICSFragmentGenerator.hpp"
#include "CDPL/Chem/BemisMurckoAnalyzer.hpp"
#include "CDPL/Chem/TopologicalEntityAlignment.hpp"
#include "CDPL/Chem/SpatialEntityAlignment.hpp"
#include "CDPL/Chem/Atom3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/AtomConformer3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/AtomArray3DCoordinatesFunctor.hpp"
#include "CDPL/Chem/UtilityFunctions.hpp"
#include "CDPL/Chem/AtomContainerFunctions.hpp"
#include "CDPL/Chem/Entity3DContainerFunctions.hpp"
#include "CDPL/Chem/AtomFunctions.hpp"
#include "CDPL/Chem/BondFunctions.hpp"
#include "CDPL/Chem/MolecularGraphFunctions.hpp"
#include "CDPL/Chem/MoleculeFunctions.hpp"
#include "CDPL/Chem/FragmentFunctions.hpp"
#include "CDPL/Chem/ReactionFunctions.hpp"

#endif // CDPL_CHEM_HPP
