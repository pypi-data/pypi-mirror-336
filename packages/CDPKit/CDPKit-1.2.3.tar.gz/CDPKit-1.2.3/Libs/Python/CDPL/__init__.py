##
# __init__.py  
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; either
# version 2 of the License, or (at your option) any later version.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Lesser General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public License
# along with this program; see the file COPYING. If not, write to
# the Free Software Foundation, Inc., 59 Temple Place - Suite 330,
# Boston, MA 02111-1307, USA.
##


from __future__ import absolute_import

import os
import sys
import platform

if sys.version_info[1] >= 8 and platform.system() == 'Windows':
    if 'site-packages' in __file__:
        os.add_dll_directory(os.path.dirname(__file__))
    else:
        os.add_dll_directory(os.path.join(os.path.dirname(os.path.dirname(os.path.dirname(__file__))), '@CDPKIT_EXECUTABLE_INSTALL_DIR@'))

from ._cdpl import *

__version__ = VERSION_STRING
