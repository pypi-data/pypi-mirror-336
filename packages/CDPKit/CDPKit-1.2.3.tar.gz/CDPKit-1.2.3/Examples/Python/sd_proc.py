#!/bin/env python

##
# sd_proc.py 
#
# This file is part of the Chemical Data Processing Toolkit
#
# Copyright (C) 2003 Thomas Seidel <thomas.seidel@univie.ac.at>
#
# Permission to use, copy, modify, and/or distribute this software for any purpose with
# or without fee is hereby granted.
# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH REGARD
# TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS. IN
# NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL
# DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER
# IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF OR IN
# CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
##


import sys

import CDPL.Chem as Chem


# retrieves and outputs the MDL structure data of the given molecule
def printStructureData(mol: Chem.Molecule) -> None: 
    if not Chem.hasStructureData(mol):        # is a structure data property available?
        print('Error: no structure data available for molecule \'%s\'' % Chem.getName(mol))
        return

    struct_data = Chem.getStructureData(mol)  # retrieve structure data
    
    print('Structure data (%s entries) of molecule \'%s\':\n' % (str(len(struct_data)), Chem.getName(mol)))

    i = 1
    
    for entry in struct_data:                 # iterate of structure data entries consisting of a header line and the actual data
        print('----- Entry %s -----' % str(i))
        print('Header: ' + entry.header)
        print('Data: ' + entry.data)

        i += 1
    
def main() -> None:
    if len(sys.argv) < 2:
        sys.exit('Usage: %s <input SD-file>' % sys.argv[0])

    # create reader for MDL SD-files
    reader = Chem.FileSDFMoleculeReader(sys.argv[1])
    
    # create an instance of the default implementation of the Chem.Molecule interface
    mol = Chem.BasicMolecule()
    
    # read and process molecules one after the other until the end of input has been reached
    try:
        while reader.read(mol): 
            try:
                printStructureData(mol)
            except Exception as e:
                sys.exit('Error: processing of molecule failed: ' + str(e))
                
    except Exception as e: # handle exception raised in case of severe read errors
        sys.exit('Error: reading molecule failed: ' + str(e))

    sys.exit(0)
        
if __name__ == '__main__':
    main()
