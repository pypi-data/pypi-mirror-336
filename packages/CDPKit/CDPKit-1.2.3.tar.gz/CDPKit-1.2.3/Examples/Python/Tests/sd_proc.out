Structure data (36 entries) of molecule '215':

----- Entry 1 -----
Header: <INCHI_IDENTIFIER>
Data: InChI=1S/C20H21FN2O/c1-23(2)11-3-10-20(17-5-7-18(21)8-6-17)19-9-4-15(13-22)12-16(19)14-24-20/h4-9,12H,3,10-11,14H2,1-2H3
----- Entry 2 -----
Header: <SYNONYMS>
Data: Nitalapram 
----- Entry 3 -----
Header: <JCHEM_RULE_OF_FIVE>
Data: 1.0
----- Entry 4 -----
Header: <DRUGBANK_ID>
Data: DB00215
----- Entry 5 -----
Header: <JCHEM_VEBER_RULE>
Data: 1.0
----- Entry 6 -----
Header: <JCHEM_PKA_STRONGEST_BASIC>
Data: 9.784978
----- Entry 7 -----
Header: <ALOGPS_SOLUBILITY>
Data: 0.00588
----- Entry 8 -----
Header: <JCHEM_AVERAGE_POLARIZABILITY>
Data: 35.39618
----- Entry 9 -----
Header: <JCHEM_IUPAC>
Data: 1-[3-(dimethylamino)propyl]-1-(4-fluorophenyl)-1,3-dihydro-2-benzofuran-5-carbonitrile
----- Entry 10 -----
Header: <JCHEM_TRADITIONAL_IUPAC>
Data: recital
----- Entry 11 -----
Header: <JCHEM_REFRACTIVITY>
Data: 94.0202
----- Entry 12 -----
Header: <JCHEM_POLAR_SURFACE_AREA>
Data: 36.26
----- Entry 13 -----
Header: <ALOGPS_LOGS>
Data: -4.74
----- Entry 14 -----
Header: <INCHI_KEY>
Data: WSEQXVZVJXJVFP-UHFFFAOYSA-N
----- Entry 15 -----
Header: <JCHEM_PHYSIOLOGICAL_CHARGE>
Data: 1.0
----- Entry 16 -----
Header: <ALOGPS_LOGP>
Data: 3.58
----- Entry 17 -----
Header: <PRODUCTS>
Data: Abbott-citalopram; Accel-citalopram Tablets; Act Citalopram; Ag-citalopram; Apo-citalopram; Auro-citalopram; Ava-citalopram; Celexa; Celexa 10 mg; Celexa 20 mg; Celexa 40 mg; Citalopram; Citalopram Hydrobromide; Citalopram Tablets; Citalopram Tablets USP; Citalopram-10; Citalopram-20; Citalopram-40; Citalopram-odan; Ctp 30; Direct Rx; Dom-citalopram; Ecl-citalopram; Ipg-citalopram; Jamp-citalopram; Manda-citalopram; Mar-citalopram; Mint-citalopram; Mylan-citalopram; Nat-citalopram; Nat-citalopram Tablets; Ng Citalopram; Novo-citalopram; Ntp-citalopram; Nu-citalopram; PHL-citalopram; PMS-citalopram; Pendo-citalopram; Perrigo Citalopram; Q-citalopram; Ran-citalo; Ran-citalopram; Ratio-citalopram; Riva-citalopram; Sandoz Citalopram; Sentralopram Am-10; Septa-citalopram; Teva-citalopram; Van-citalopram
----- Entry 18 -----
Header: <JCHEM_LOGP>
Data: 3.7643042
----- Entry 19 -----
Header: <DATABASE_NAME>
Data: drugbank
----- Entry 20 -----
Header: <DRUG_GROUPS>
Data: approved
----- Entry 21 -----
Header: <SMILES>
Data: CN(C)CCCC1(OCC2=C1C=CC(=C2)C#N)C1=CC=C(F)C=C1
----- Entry 22 -----
Header: <EXACT_MASS>
Data: 324.1638
----- Entry 23 -----
Header: <JCHEM_ROTATABLE_BOND_COUNT>
Data: 5.0
----- Entry 24 -----
Header: <JCHEM_MDDR_LIKE_RULE>
Data: 0.0
----- Entry 25 -----
Header: <JCHEM_GHOSE_FILTER>
Data: 1.0
----- Entry 26 -----
Header: <JCHEM_NUMBER_OF_RINGS>
Data: 3.0
----- Entry 27 -----
Header: <GENERIC_NAME>
Data: Citalopram
----- Entry 28 -----
Header: <JCHEM_BIOAVAILABILITY>
Data: 1.0
----- Entry 29 -----
Header: <JCHEM_ACCEPTOR_COUNT>
Data: 3.0
----- Entry 30 -----
Header: <FORMULA>
Data: C20H21FN2O
----- Entry 31 -----
Header: <MOLECULAR_WEIGHT>
Data: 324.3919
----- Entry 32 -----
Header: <DATABASE_ID>
Data: DB00215
----- Entry 33 -----
Header: <SALTS>
Data: Citalopram Hydrobromide
----- Entry 34 -----
Header: <JCHEM_DONOR_COUNT>
Data: 0.0
----- Entry 35 -----
Header: <INTERNATIONAL_BRANDS>
Data: Akarin; Celapram; Ciazil; Cilift; Cipram; Cipramil; Ciprapine; Citabax; Citalec; Citol; Citopam; Citox; Citrol; Dalsan; Elopram; Humorup; Oropram; Pramcit; Recital; Seropram; Talam; Talohexal; Temperax; Vodelax; Zentius; Zetalo
----- Entry 36 -----
Header: <JCHEM_FORMAL_CHARGE>
Data: 0.0
